core/model-api/src/main/java/org/eclipse/rdf4j/model/base/AbstractBNode.java

```diff
@@ -22,6 +23,37 @@
 
 	private static final long serialVersionUID = -437354568418943981L;
 
+	private static final AtomicLong nodeID = new AtomicLong(ThreadLocalRandom.current().nextLong());
+
+	/**
+	 * Creates a new blank node value.
+	 *
+	 * @return a new generic blank node value with a system-generated label
+	 */
+	public static BNode createBNode() {
+		return new GenericBNode(Long.toHexString(Math.abs(nodeID.getAndIncrement())));
+	}
+
+	/**
+	 * Creates a new blank node value.
+	 *
+	 * @param nodeID the identifier of the blank node
+	 *
+	 * @return a new generic blank node value
+	 *
+	 * @throws NullPointerException if {@code nodeID} is {@code null}
+	 */
+	public static BNode createBNode(String nodeID) {
```

Comment:
Why are we adding factory methods to the class itself? This will be problematic to support. For example concrete subclasses will have their own factories, but if you do `MyConcreteBNode.createBNode` you suddenly get a `GenericBNode` instance again. It also somewhat defeats the purpose of having a separate ValueFactory.